### PR TITLE
Set max buffer for kryo to serialize

### PR DESCRIPTION
**Pull request by rerorero**
## WHAT
This PR stops using ByteArrayOutputStream to serialize and enable to specify Output buffersize.

## WHY
There is a problem with incorrect size of serialized data using ByteArrayOutputStream with default max buffersize of Output.
Let's use the buffer managed by Output itself.

### CODE REVIEW

File: kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/CipherField.java [hunk 1]
```diff
@@ -105,6 +107,8 @@
   private static final String KEY_SOURCE_DEFAULT = "CONFIG";
   private static final String CIPHER_DATA_KEYS_DEFAULT = "[]";
   private static final String KMS_KEY_NAME_DEFAULT = null;
+  private static final int KRYO_OUTPUT_BUFFER_SIZE_DEFAULT = 32;
```
Comment (rerorero) [inline kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/CipherField.java:110]: This is same as the default size of ByteArrayOutputStream
https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/java/io/ByteArrayOutputStream.java#L64

File: kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/CipherField.java [hunk 2]
```diff
@@ -207,7 +211,19 @@
...
+              Type.INT,
+              KRYO_OUTPUT_BUFFER_SIZE_DEFAULT,
+              Importance.LOW,
+              "Initial buffer size for kryo to serialize. Default is 32")
+          .define(
+              KRYO_OUTPUT_BUFFER_SIZE_MAX,
+              Type.INT,
+              KRYO_OUTPUT_BUFFER_SIZE_MAX_DEFAULT,
+              Importance.LOW,
+              "Maximum buffer size for kryo to serialize. Default -1 corresponds to no upper limit (up to Integer.MAX_VALUE - 8 technically).");
```
Comment (rerorero) [inline kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/CipherField.java:226]: ref. https://github.com/EsotericSoftware/kryo/blob/7b640984413c5e2bbed23ef3e5eb20204f79f94e/src/com/esotericsoftware/kryo/util/Util.java#L57:26

File: kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/DataKeyConfig.java [hunk 3]
```diff
@@ -74,9 +74,6 @@ public int hashCode() {
 
   @Override
   public String toString() {
-    return "DataKeyConfig{"
-        + "identifier='"
-        + getIdentifier()
-        + "'}";
+    return "DataKeyConfig{" + "identifier='" + getIdentifier() + "'}";
```
Comment (rerorero) [inline kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/DataKeyConfig.java:77]: This was changed by `make fmt`, not related with this PR.

File: kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/RecordHandler.java [hunk 4]
```diff
@@ -82,12 +82,7 @@ public Object processField(Object object, String matchedPath) {
       }
     } catch (Exception e) {
       throw new DataException(
-          "error: "
-              + cipherMode
-              + " of field path '"
-              + matchedPath
-              + "' failed unexpectedly",
-          e);
+          "error: " + cipherMode + " of field path '" + matchedPath + "' failed unexpectedly", e);
```
Comment (rerorero) [inline kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/RecordHandler.java:85]: This was changed by `make fmt`, not related with this PR.

**Review by darklore — APPROVED** [review]

File: kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/serdes/KryoSerdeProcessor.java [hunk 5]
```diff
@@ -34,15 +33,20 @@
...
 
   private static final Logger LOGGER = LoggerFactory.getLogger(KryoSerdeProcessor.class);
+  private final int outputBufSize;
+  private final int outputBufMaxSize;
 
-  public KryoSerdeProcessor() {}
+  public KryoSerdeProcessor(int outputBufSz, int outputBufMaxSz) {
+    this.outputBufSize = outputBufSz;
+    this.outputBufMaxSize = outputBufMaxSz;
+  }
```
Comment (laughingman7743) [inline kafka-connect-transform-kryptonite-gcp/src/main/java/com/github/hpgrahsl/kafka/connect/transforms/kryptonite/serdes/KryoSerdeProcessor.java:42]:
```suggestion
  public KryoSerdeProcessor(int outputBufSize, int outputBufMaxSize) {
    this.outputBufSize = outputBufSize;
    this.outputBufMaxSize = outputBufMaxSize;
  }
```

**Review by laughingman7743 — APPROVED** [review]

**Pull request merged**